### DISCUSSION TITLE
Data analysis using local pc

### DISCUSSION THREAD

**Discussion by dpkhatua** [epochpic/epoch#563] [General]
Hi All,
Can anyone suggest me how to handle the .sdf files using local pc? can we open them using python/matlab and plot all the different data using personal laptop or PC?

Thanks,

**Comment by Status-Mirror** [comment]
Hey @Khatua03

Someone in our group wrote a Python package, which is documented [here](https://epochpic.github.io/documentation/visualising_output/python.html).

I prefer to use MATLAB - there is a function in `epoch/SDF/Matlab` called `GetDataSDF.m`. If you add all the functions in this folder to your path, you should be able to run `data= GetDataSDF('0000.sdf');` to create a MATLAB structure with all the data from the file.

Hope this helps,
Stuart

**Reply by dpkhatua** [reply]
Thanks Stuart.
I guess this python module is only for linux based system. Is there any way to use it in window operating systems? I want to store the sdf output files in window and use it for analysis.

Thanks.

**Reply by DanRRRR** [reply]
I also use Windows for processing of EPOCH raw data extracted as usual text or binary files instead of just the SDF files. Though Matlab or Visit are great software i find using them with PIC codes not convenient and for that i use my own 3D software which with just a click visualizes raw data from any PIC codes people created or will create in the future. All that is done Instantly as soon as their data is generated because all is written purely in Fortran which reliably loads standard ASCII/binary format PIC data  by all the software in the world but in this case with blazing speed up to 6-7 GB per second Fortran is particularly good at. Lose on data load just 10 minutes per day and at the end of life you will lose full 1 years just waiting for the data load. With 3D data i am sure we lose much much more than that

I installed Wine in Linux  which brings me Windows as one of Linux usual windows. With that all Windows commands and most known Windows programs will work either like this is itself is Windows and not Linux or via additional word "wine" preceding it in Terminal (Command Prompt). Also for work with files i use mostly another great software called Double Commander (and in Windows part also Total Commander) more than Terminal or Files

Would be great if some users or developers made SDF file converter to ASCII/binary data, with that most of the problems for everyone in visualization of PIC code data would be solved

**Reply by Status-Mirror** [reply]
Hey @Khatua03,

Unfortunately, I'm not the python guy. It probably could be made to work with Windows, but I wouldn't know how to go about making those changes. We're currently in the process of upgrading EPOCH, and this will involve more Python support, but I wouldn't expect this soon.

Apologies for now,
Stuart

**Reply by dpkhatua** [reply]
Hi @Status-Mirror Thanks. 
I will try to run it in window.

Regards,
Durga Prasad Khatua.
